pyodide-build/pyodide_build/io.py

```diff
@@ -126,6 +127,13 @@ def _check_config_build(config: dict[str, Any]) -> Iterator[str]:
     build_metadata = config["build"]
     library = build_metadata.get("library", False)
     sharedlibrary = build_metadata.get("sharedlibrary", False)
+    exports = build_metadata.get("exports", "pyinit")
+    if isinstance(exports, str) and exports not in [
+        "pyinit",
+        "requested",
+        "whole_archive",
+    ]:
+        yield f"build/exports must be 'pyinit', 'explicit', 'all', or a list of strings not {build_metadata['exports']}"
```

Comment:
This doesn't seem to match with docs?

Reply:
Yeah I left the list of strings case undocumented since we aren't actually using it anywhere currently. We might want to take it out, or when we first use it we can document it. It seemed to work okay when I tried it but since the CI doesn't use it it's hard to know if it really works. I thought about explicitly listing the exports for xgboost but there are quite a lot.

Reply:
Oh I see what you mean...

Reply:
Oh I see what you mean, the error message is a bit wrong.

Reply:
#2777 fixes it.